[ROCm] torch/adapter_v2.cc requring cuda_runtime_api.h on ROCm builds

**Environment:**
1. Framework: PyTorch
2. Framework version: v1.12.1_src
3. Horovod version: v0.26.1
4. MPI version: MVAPICH2-GDR-v2.3.7-ROCm
5. ROCm version: 5.1.1
6. NCCL version: N/A
7. Python version: 3.9.16
8. Spark / PySpark version: N/A
9. Ray version: N/A
10. OS and version: Rocky Linux release 8.5 (Green Obsidian)
11. GCC version: 10.3.0
12. CMake version: 3.22.2

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.

Compilation Error:

Buiding `horovod/torch/CMakeFiles/pytorch.dir/adapter_v2.cc.o` requires `cuda_runtime_api.h`

```
[ 97%] Building CXX object horovod/torch/CMakeFiles/pytorch.dir/adapter_v2.cc.o
cd /home/xu.3304/horovod/build/temp.linux-x86_64-cpython-39/RelWithDebInfo/horovod/torch && /home/xu.3304/mvapich2-install/amd/gdr2.3.7_rocm5.1.1_gcc10.3.0/bin/mpicxx -DEIGEN_MPL2_ONLY=1 -DHAVE_GPU=1 -DHAVE_MPI=1 -DHAVE_ROCM=1 -DHOROVOD_GPU_ALLREDUCE=77 -DPYTORCH_VERSION=1012000000 -DTORCH_API_INCLUDE_EXTENSION_H=1 -Dpytorch_EXPORTS -I/home/xu.3304/horovod/third_party/HTTPRequest/include -I/home/xu.3304/horovod/third_party/boost/assert/include -I/home/xu.3304/horovod/third_party/boost/config/include -I/home/xu.3304/horovod/third_party/boost/core/include -I/home/xu.3304/horovod/third_party/boost/detail/include -I/home/xu.3304/horovod/third_party/boost/iterator/include -I/home/xu.3304/horovod/third_party/boost/lockfree/include -I/home/xu.3304/horovod/third_party/boost/mpl/include -I/home/xu.3304/horovod/third_party/boost/parameter/include -I/home/xu.3304/horovod/third_party/boost/predef/include -I/home/xu.3304/horovod/third_party/boost/preprocessor/include -I/home/xu.3304/horovod/third_party/boost/static_assert/include -I/home/xu.3304/horovod/third_party/boost/type_traits/include -I/home/xu.3304/horovod/third_party/boost/utility/include -I/home/xu.3304/horovod/third_party/lbfgs/include -I/home/xu.3304/horovod/third_party/eigen -I/home/xu.3304/horovod/third_party/flatbuffers/include -isystem /opt/rocm-5.1.1/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include/TH -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include/THC -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include/THH -isystem /opt/rocm-5.1.1/miopen/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/include/python3.9 -DCUDA_HAS_FP16=1 -D__HIP_NO_HALF_OPERATORS__=1 -D__HIP_NO_HALF_CONVERSIONS__=1 -D_GLIBCXX_USE_CXX11_ABI=1 -D__HIP_PLATFORM_HCC__=1  -pthread -fPIC -Wall -ftree-vectorize -mf16c -mavx -mfma -O3 -g -DNDEBUG -fPIC -std=c++14 -MD -MT horovod/torch/CMakeFiles/pytorch.dir/adapter_v2.cc.o -MF CMakeFiles/pytorch.dir/adapter_v2.cc.o.d -o CMakeFiles/pytorch.dir/adapter_v2.cc.o -c /home/xu.3304/horovod/horovod/torch/adapter_v2.cc
In file included from /home/xu.3304/horovod/horovod/torch/adapter_v2.cc:17:
/home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.26.1-mv2gdr2.3.7rocm/lib/python3.9/site-packages/torch/include/c10/cuda/CUDAStream.h:6:10: fatal error: cuda_runtime_api.h: No such file or directory
    6 | #include <cuda_runtime_api.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
Note: This is a pure ROCm build
1. MVAPICH2-GDR backend is built from source with only ROCm support
2. Pytorch v1.12.1 is built from source with MVAPICH2-GDR as backend and is a pure ROCm build
3. Since all components are built from ROCm, should the condition change when checking if there's a CUDA header? There won't be a `cuda_runtime_api.h` for a ROCm built pytorch. Please correct me if I am wrong.
4. I noticed that the file is hipified but the version didn't get picked up during building
```
/home/xu.3304/horovod/horovod/torch/adapter_v2.cc -> /home/xu.3304/horovod/horovod/torch/adapter_v2_hip.cc [ok]
```

Horovod build from source command: 
```
HOROVOD_ROCM_HOME=/opt/rocm-$ROCM_VERSION \
CC=$(which mpicc) \
CXX=$(which mpicxx) \
HOROVOD_WITHOUT_GLOO=1 \
HOROVOD_WITH_MPI=1 \
HOROVOD_GPU=ROCM \
HOROVOD_GPU_ALLREDUCE=MPI \
HOROVOD_WITHOUT_TENSORFLOW=1 \
HOROVOD_WITH_PYTORCH=1 \
HOROVOD_WITHOUT_MXNET=1 \
/home/xu.3304/miniconda3/envs/$CONDA_ENV/bin/python setup.py install
```
I have attached a full horovod install log.
[hvd0.26.1_torch1.12.1_gdr2.3.7_rocm5.1.1_gcc10.3.0.log](https://github.com/horovod/horovod/files/10572012/hvd0.26.1_torch1.12.1_gdr2.3.7_rocm5.1.1_gcc10.3.0.log)

--- a/horovod/torch/CMakeLists.txt
+++ b/horovod/torch/CMakeLists.txt
@@ -74,7 +74,7 @@ list(APPEND PYTORCH_SOURCES "${PROJECT_SOURCE_DIR}/horovod/torch/handle_manager.
                             "${PROJECT_SOURCE_DIR}/horovod/torch/ready_event_hip.cc"
                             "${PROJECT_SOURCE_DIR}/horovod/torch/hip_util.cc"
                             "${PROJECT_SOURCE_DIR}/horovod/torch/mpi_ops_v2_hip.cc"
-                            "${PROJECT_SOURCE_DIR}/horovod/torch/adapter_v2.cc")
+                            "${PROJECT_SOURCE_DIR}/horovod/torch/adapter_v2_hip.cc")
 else()
 list(APPEND PYTORCH_SOURCES "${PROJECT_SOURCE_DIR}/horovod/torch/handle_manager.cc"
                             "${PROJECT_SOURCE_DIR}/horovod/torch/ready_event.cc"

